src/workflows/client/client.py

```diff
@@ -0,0 +1,288 @@
+        context: Union[Context, dict[str, Any], None] = None,
+        **kwargs: Any,
+    ) -> Handler:
+        if start_event and isinstance(start_event, StartEvent):
+            start_event = start_event.model_dump()
+        if context and isinstance(context, Context):
+            try:
+                context = context.to_dict()
+            except Exception as e:
+                raise ValueError(f"Impossible to serialize context because of: {e}")
+        response = await post_workflows_name_run_nowait(
+            name=workflow_name,
+            client=self._client,
+            body=PostWorkflowsNameRunNowaitBody(
+                start_event=PostWorkflowsNameRunNowaitBodyStartEvent.from_dict(
+                    cast(dict, start_event) or {}
+                ),
+                context=PostWorkflowsNameRunNowaitBodyContext.from_dict(context or {}),
+                kwargs=PostWorkflowsNameRunNowaitBodyKwargs.from_dict(kwargs),
+            ),
+        )
+        if isinstance(response, Handler):
+            return response
+        else:
+            raise ValueError("Response was not properly generated")
+
+    async def get_workflow_events(
+        self,
+        handler: Handler,
+    ) -> AsyncGenerator[dict[str, Any], None]:
+        """
+        Stream events using newline-delimited JSON format
+        """
+        url = f"/events/{handler.handler_id}?sse=false"
+        client = self._client.get_async_httpx_client()
+        try:
+            async with client.stream("GET", url) as response:
+                # Handle different response codes
+                if response.status_code == 404:
+                    raise ValueError("Handler not found")
+                elif response.status_code == 204:
+                    # Handler completed, no more events
+                    return
+
+                response.raise_for_status()
+
+                async for line in response.aiter_lines():
+                    if line.strip():  # Skip empty lines
+                        try:
+                            event = json.loads(line.replace("\n", ""))
+                            yield event
+                        except json.JSONDecodeError as e:
+                            print(f"Failed to parse JSON: {e}, data: {line}")
+                            continue
+
+        except httpx.TimeoutException:
+            raise TimeoutError(
+                f"Timeout waiting for events from handler {handler.handler_id}"
+            )
+        except httpx.RequestError as e:
+            raise ConnectionError(f"Failed to connect to event stream: {e}")
+
+    async def get_workflow_handlers(self) -> list[Handler]:
+        response = await get_handlers(client=self._client)
+        if response:
+            return response.handlers
+        else:
+            raise ValueError("Response was not properly generated")
+
+    async def get_workflow_result(self, handler: Handler) -> Any:
```

Comment:
shouldn't this just be a handler id? as input, and return a Handler as a result?